packages/block-library/src/image/edit.js

```diff
@@ -352,6 +360,13 @@ class ImageEdit extends Component {
 		} ) );
 	}
 
+	getImageSizeByURL( url ) {
```

Comment:
I am thinking it might be nicer to get rid of this function altogether and instead replace `updateImageURL()` with a function like `updateImage()` that takes `slug` instead of `URL`. 

https://github.com/WordPress/gutenberg/compare/update/image-classes-with-dimensions...try/updateImage-instead-of-updateImageURL

Reply:
For this to work we'd need to change what the Selector component stores for values, size slugs instead of URLs.

Reply:
Is that a problem? `SelectControl` ought to not care what you give it as a `value`.